src/hooks/useShortcut.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useCallback, useReducer } from 'react'
+
+const keysReducer = (state, action) => {
+  switch (action.type) {
+    case 'set-key-down':
```

Comment:
You could think about upper case constants for your reducer types:

```suggestion
    case SET_KEY_DOWN:
```
(Discussion for Redux, but it makes sense here too)
https://redux.js.org/style-guide/style-guide/#write-action-types-as-domaineventname
https://stackoverflow.com/questions/34965856/what-is-the-point-of-the-constants-in-redux

https://javascript.info/variables#uppercase-constants

Reply:
Thank you for the Input, i will change it and keep it in mind!